lnbits/wallets/alby.py

```diff
@@ -71,57 +83,104 @@ async def create_invoice(
         else:
             data["memo"] = memo or ""
 
-        r = await self.client.post(
-            "/invoices",
-            json=data,
-            timeout=40,
-        )
-
-        if r.is_error:
-            error_message = r.json()["message"]
-            return InvoiceResponse(False, None, None, error_message)
-
-        data = r.json()
-        checking_id = data["payment_hash"]
-        payment_request = data["payment_request"]
-        return InvoiceResponse(True, checking_id, payment_request, None)
+        try:
+            r = await self.client.post(
+                "/invoices",
+                json=data,
+                timeout=40,
+            )
+            r.raise_for_status()
+
+            data = r.json()
+
+            if r.is_error:
+                error_message = data["message"]
+                return InvoiceResponse(False, None, None, error_message)
+
+            checking_id = data["payment_hash"]
+            payment_request = data["payment_request"]
+            return InvoiceResponse(True, checking_id, payment_request, None)
+        except KeyError as exc:
+            logger.warning(exc)
+            return InvoiceResponse(
+                False, None, None, "Server error: 'missing required fields'"
+            )
+        except json.JSONDecodeError as exc:
+            logger.warning(exc)
+            return InvoiceResponse(
+                False, None, None, "Server error: 'invalid json response'"
+            )
+        except Exception as exc:
+            logger.warning(exc)
+            return InvoiceResponse(
+                False, None, None, f"Unable to connect to {self.endpoint}."
+            )
 
     async def pay_invoice(self, bolt11: str, fee_limit_msat: int) -> PaymentResponse:
-        # https://api.getalby.com/payments/bolt11
-        r = await self.client.post(
-            "/payments/bolt11",
-            json={"invoice": bolt11},  # assume never need amount in body
-            timeout=None,
-        )
-
-        if r.is_error:
-            error_message = r.json()["message"]
-            return PaymentResponse(False, None, None, None, error_message)
-
-        data = r.json()
-        checking_id = data["payment_hash"]
-        fee_msat = -data["fee"]
-        preimage = data["payment_preimage"]
-
-        return PaymentResponse(True, checking_id, fee_msat, preimage, None)
+        try:
+            # https://api.getalby.com/payments/bolt11
+            r = await self.client.post(
+                "/payments/bolt11",
+                json={"invoice": bolt11},  # assume never need amount in body
+                timeout=None,
+            )
+            r.raise_for_status()
+            data = r.json()
+
+            if r.is_error:
+                error_message = data["message"]
+                return PaymentResponse(False, None, None, None, error_message)
+
+            checking_id = data["payment_hash"]
+            # todo: confirm with bitkarrot that having the minus is fine
```

Comment:
cc: @bitkarrot ☝️

Reply:
the fee value returned by Alby is positive.